CHANGELOG.md

```diff
@@ -4,6 +4,244 @@ All notable changes to this project will be documented in this file.
+  - Several key types can be previewed directly in the track editor.
+  - Tracks can now be grouped and filtered on a per-node basis.
+  - Copying and pasting tracks between animations is now possible.
+  - New Capture mode to blend from a node's current value to the first key in a track.
+- [Improved animation tree and new state machine.](https://godotengine.org/article/godot-gets-new-animation-tree-state-machine)
+  - More visual feedback in the blend tree editor.
+  - 1D and 2D blend spaces are now supported.
+  - Ability to write custom blending logic.
+  - Support for root motion.
+- [New FileSystem dock.](https://godotengine.org/article/godot-gets-new-filesystem-dock-3-1)
+  - Unified view of folders and files in the same panel.
+  - Files can now be marked as favorites, not only folders.
+  - Files now have icons representing their type, or thumbnail previews when relevant.
+  - New search field to filter entries in the tree.
+- [OpenSimplexNoise and NoiseTexture resources.](https://godotengine.org/article/simplex-noise-lands-godot-31)
+- [Optional static typing in GDScript.](https://godotengine.org/article/optional-typing-gdscript)
+  - Does not currently improve performance, but helps write more robust code.
+- Warning system in GDScript.
+  - Reports potential code issues such as:
+    - unused variables,
+    - standalone expressions,
+    - discarded return values from functions,
+    - unreachable code after a `return` statement,
+    - …
+  - Warnings can be disabled in the Project Settings or by writing special comments.
+- [GDScript keyword `class_name` to register scripts as classes.](http://docs.godotengine.org/en/latest/getting_started/step_by_step/scripting_continued.html#register-scripts-as-classes)
+- Simple expression language independent from GDScript, used by inspector boxes that accept numeric values.
+  - Can also be used in projects.
+- C# projects can now be exported for Windows, Linux, and macOS targets.
+- The `server` platform is back as it was in Godot 2.1.
+  - It is now again possible to run a headless Godot instance on Linux.
+- Support for BPTC texture compression on desktop platforms.
+- New properties for SpatialMaterial.
+  - Dithering-based distance fade, for fading materials without making them transparent.
+  - Disable ambient light on a per-material basis.
+- Option to link Mono statically on Windows.
+- Revamped TileSet editor.
+- RandomNumberGenerator class that allows for multiple instances at once.
+- Array methods `min()` and `max()` to return the smallest and largest value respectively.
+- Dictionary method `get(key[, default])` where `default` is returned if the key does not exist.
+- Node method `print_tree_pretty()` to print a graphical view of the scene tree.
+- String methods `trim_prefix()`, `trim_suffix()`, `lstrip()`, `rstrip()`.
+- OS method `get_system_time_msecs()` to return the system time with milliseconds.
+- OS method `center_window()` to center the window on the screen.
```

Comment:
Some missing OS stuff:

- `OS::get_video_driver_count`, `OS::get_video_driver_name` - video driver enumeration
- `OS::get_audio_driver_count`, `OS::get_audio_driver_name` - audio driver enumeration
- `OS::get_connected_midi_inputs`, `OS::open_midi_inputs`, `OS::close_midi_inputs` (and related drivers) - MIDI hardware support, #20154
- `OS::window_per_pixel_transparency_enabled` - window / splash-screen transparency support, #14622
- `OS::move_window_to_foreground` exposed to GDScript